internal/action/action.go

```diff
@@ -2,18 +2,171 @@ package action
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/Minecraft-Unified-Hub-Team/ServerControl/utils/mine_os"
+	"github.com/Minecraft-Unified-Hub-Team/ServerControl/utils/mine_state"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	cd            = "cd"
+	run           = "run.sh"
+	serverPath    = "/server"
+	baseURL       = "https://maven.minecraftforge.net/net/minecraftforge/forge/%s"
+	installerName = "/forge-%s-installer.jar"
 )
 
-type ActionInterface interface {
-	Start(context.Context) error
+type ActionService struct {
+	AliveCtx context.Context    // context that continues until server is stopped or dead
+	stopCtx  context.CancelFunc // function that cancels server binary execution
+
+	State *mine_state.State // channel that stores state of server
 }
 
 func NewActionService() (*ActionService, error) {
-	return &ActionService{}, nil
+	currentState, _ := mine_state.NewState(mine_state.Stopped) // TODO set mine_state.Stopped here when install will be complited
+	return &ActionService{
+		State: currentState,
+	}, nil
+}
+
+func (as *ActionService) downloadJar(ctx context.Context, version string) error {
+	var err error = nil
+	var errorFormat string = fmt.Sprintf("ActionService.downloadJar(ctx, %s)", version) + ": %w"
+
+	/* configure url address for downloading correct forge jar */
```

Comment:
Should we really leave comments like that? For me, the code seems very obvious and doesn't need additional explanations.

Reply:
will fix in future